### DISCUSSION TITLE
Can I use github to develop a story (instead of software)?? <title>

### DISCUSSION THREAD

**Discussion by tsmspace** [community/community#46392] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello. 

I am a big fan of github, and although I am not a software developer, I have been an arduino enthusiast and github has been an amazing place, with one of the projects having changed my life immensely, allowing me to easily make a game controller with unique features that are not available from any product. This game controller project (viewable at tsmspace.com) has given me big dreams. The original design is as a drone controller for a drone such as the NASA SPHERES project, to be used for racing FPV. I imagine people launching courses of inflatable hoops and piloting drones around them as the first real sport in space. 

As an effect of my gaming with my game controller, I have come to want to work on a science fiction story, where i have some rules, but invite others to help me think of plots and characters that might go well in the story. This story will hopefully impact or inspire the development of a simulator for space drone racing. I would like to try and use the github engine / profile style to make a sort of "github of stories", where people can "fork" existing stories, and make changes or contributions to them so that the original and other forks are all available to anyone who wishes to explore them. So much about the organization of github is perfect, especially the concept of open source material. 

Would it be ok if I made a project for this purpose? Nothing will be code, but instead the repositories will be bits and pieces of story development. Then, I would be able to post all of the bits of outline I have made so far, as well as invite others to both contribute to and fork the project. 

I suppose I could just do it and see how it goes, but I wanted to ask about it first??? 

Thanks! 

again,, tsmspace.com 
special thanks to cleric-K for the vjoyserialfeeder project, which has made so much of my last several years possible.

**Comment by chris-c-thomas** [comment]
You can absolutely do something like that! What file format were you thinking you'd write it in? Code files are just text based files at the end of the day. Documentation websites are basically just large books as well. 

If you keep binary type files that will work too but in a little more limited capacity. The individual changes to the file won't be recorded just the fact that the file as a whole changed.

Other options besides just here exist such as [GitBook](https://www.gitbook.com/) that is pretty popular. They have a free and paid tiers.

**Reply by tsmspace** [reply]
I will investigate both , then, and probably try both. Gitbook does look like even though it is still not the function of it to make stories,, people who are not familiar with github will be more comfortable right away.

**Comment by henrich45** [comment]
Using GitHub to organize and develop collaborative storytelling in the same way open-source projects are managed is a unique approach. While GitHub is primarily designed for code, it can definitely handle non-code projects as well, such as writing and story development. Repositories can be used for outlines, plots, and character development, while forks and contributions can encourage collaborative growth, just like with software[.](https://wormzonemodi.com/)

You can absolutely create a project for this purpose. Simply set up your repository, organize the story elements into separate files or folders, and invite contributors. Using GitHub’s version control for stories will let you keep track of different ideas and story arcs. It's an innovative way to foster collaboration and creativity—good luck!

**Comment by ebndev** [comment]
Hi @tsmspace, thanks for being a part of the GitHub Community!

That sounds like a super cool idea 😄

**Comment by thomassiard** [comment]
1. Create a Repository
You can start by creating a new repository specifically for your story project. Use a descriptive name that reflects its purpose, such as space-drone-racing-stories or similar.
2. Define Guidelines
Establish clear contribution guidelines to help collaborators understand how they can participate. This could include:
How to fork and contribute to stories.
Formatting for story submissions or plot ideas.
Any specific themes or rules you want contributors to follow.
3. Organize the Content
Use the repository structure to organize different elements of your story, such as:
Outlines: Include your initial outlines in separate markdown files.
Characters: Create a directory for character development, allowing others to add or modify character profiles.
Plot Ideas: Encourage contributors to create new branches for their plot ideas.
4. Invite Collaboration
Share your project on social media, forums, and communities related to storytelling or science fiction to invite contributions.
Encourage discussions in issues or pull requests where contributors can suggest changes or ask for feedback.
5. Iterate and Evolve
As contributions come in, be open to feedback and iterations. This could lead to exciting developments and creative directions you might not have considered.
6. Respect Copyright and Licenses
Make sure to include a license that clarifies how your content can be used or modified by others. A Creative Commons license might be suitable for a project like this.
